src/apps/filters/h3ToGeo.c

```diff
@@ -72,46 +66,68 @@ void doCell(H3Index h, int isKmlOut) {
 }
 
 int main(int argc, char *argv[]) {
-    // check command line args
-    if (argc > 5) {
-        fprintf(stderr, "usage: %s [outputMode kmlName kmlDesc]\n", argv[0]);
-        exit(1);
-    }
+    H3Index index = 0;
+    char userKmlName[BUFF_SIZE] = {0};
+    char userKmlDesc[BUFF_SIZE] = {0};
 
-    int isKmlOut = 0;
-    if (argc > 1) {
-        if (!sscanf(argv[1], "%d", &isKmlOut))
-            error("outputMode must be an integer");
+    Arg helpArg = {.names = {"-h", "--help"},
+                   .helpText = "Show this help message."};
+    Arg indexArg = {
+        .names = {"-i", "--index"},
+        .scanFormat = "%" PRIx64,
+        .valueName = "index",
+        .value = &index,
+        .helpText =
+            "Index, or not specified to read indexes from standard in."};
+    Arg kmlArg = {.names = {"-k", "--kml"},
+                  .helpText = "Print output in KML format."};
+    Arg kmlNameArg = {.names = {"-kn", "--kml-name"},
+                      .scanFormat = "%255c",  // BUFF_SIZE - 1
+                      .valueName = "name",
+                      .value = &userKmlName,
+                      .helpText = "Name of the KML file."};
```

Comment:
This isn't correct, right? Isn't this name used inside a tag in KML, rather than the file name?

Reply:
I had some difficulty with this because the tag is named "name" rather than "title", which I felt would be clearly not the name of the file. I think rewriting it to mention the tag and not refer to a "file" is a good idea.